controller/loginController.js

```diff
@@ -1,10 +1,51 @@
 const bcrypt = require('bcryptjs');
 const jwt = require('jsonwebtoken');
-let getUserCredentials = require('../model/getUserCredentials.js')
+let getUserCredentials = require('../model/getUserCredentials.js');
+let addMfaToken = require('../model/addMfaToken.js');
+const sgMail = require('@sendgrid/mail');
+
 
 const login = async (req, res) => {
     const { username, password } = req.body;
     
+    try {
+        if (!username || !password) {
+            return res.status(400).json({ error: 'Username and password are required' });
+         }
+        const user = await getUserCredentials(username, password);
+        if (user.length === 0) {
```

Comment:
Checking against length here will throw an error when user is null which will occur when the user doesn't exist in the db.
The getUserCredentials function returns the element in index 0 and doesn't check if the array actually has any elements, so that might need to be modified as well